src/Compilers/Core/Portable/PEWriter/MetadataWriter.cs

```diff
@@ -3486,7 +3502,7 @@ private void SerializeMetadataExpression(LiteralEncoder encoder, IMetadataExpres
                 }
                 else
                 {
-                    scalarEncoder.SystemType(((MetadataTypeOf)expression).TypeToGet.GetSerializedTypeName(Context));
+                    scalarEncoder.SystemType(((MetadataTypeOf)expression).TypeToGet.GetSerializedTypeName(GetEmitContextForAttribute(customAttribute)));
```

Comment:
>GetEmitContextForAttribute(customAttribute) [](http://example.com/codeflow?start=106&length=43)

Why not pass specific context from the "top" rather than creating it over and over at the "bottom".

Reply:
I wasn't sure which approach is better. But getting the syntax node is probably an expensive operation, so I guess you're right. Changed the code to pass `EmitContext` instead. I have used `in` parameters to avoid copying the large structure.